text/0000-mfa-rollout.md

```diff
@@ -0,0 +1,129 @@
+
+Example message for gem signin (general audience it can also be targeted to the specific audience)
+```
+For protection of your account and gems, we encourage you to set up multi-factor authentication at https://rubygems.org/multifactor_auth/new.
+```
+
+Example message for critical gem commands (`gem push/yank`, and `gem owner -a/r`) for the 100 most downloaded gem accounts
+```
+[WARNING] For protection of your account and gems, we encourage you to set up multi-factor authentication at https://rubygems.org/multifactor_auth/new. Your account will be required to have MFA enabled in the future.
+```
+
+## Phase 3: Enforcing MFA on Most Downloaded Gems (est. Fall 2022)
+
+At a future date, yet to be determined, we will require MFA to be enabled in order to interact with Rubygems.org. Enforcement will include the following changes.
+
+On the UI, after login, the setup MFA page will be displayed for users to setup MFA. They are able to back out, but the user will be directed to set up MFA whenever they choose to visit a page that requires authentication. A banner will also appear describing the intent. 
+
+Example message
+```
+For protection of your account and your gems, you are required to set up multi-factor authentication.
+```
+
+On the CLI, on `gem signin`, `gem push/yank`, and `gem owner -a/r`, an error message will display to require users to setup MFA on rubygems.org and will block the request from succeeding. 
+
+Example message
+```
+For protection of your account and your gems, you are required to set up multi-factor authentication at https://rubygems.org/multifactor_auth/new
+```
+
+## Phase 4: Rolling out to the General Public (2023)
+
+Once the rollout of the top 100 gems is successfully rolled out, a similar rollout will be done to another cohort of gems until all accounts are required to have MFA enabled. 
+
+# Reference-level explanation
+
+## API Key and Gem Publishing Improvements
+The API key model will need to be changed in order to implement these improvements. For new features like adding gem scope and expiry date on gems, columns will be added for expiry and gem scope and additional options will need to be added to the CLI and the UI. Setting an expiry on existing keys can be considered a breaking change.
+
+## Policy Rollout
+A group of proofs of concept are listed below that demonstrate what areas of the Rubygems codebase will be modified in order to implement the policy. More detailed explanations and implementation alternatives are available in their PR descriptions.
+
+## Recommending MFA on Most Downloaded Gems
+- UI: https://github.com/Shopify/rubygems.org/pull/3
+- CLI Rubygems.org: https://github.com/Shopify/rubygems.org/pull/9 
+- CLI Rubygems: https://github.com/Shopify/rubygems/pull/1 
+
+## Enforcing MFA on Most Downloaded Gems
+- UI: https://github.com/Shopify/rubygems.org/pull/2 
+- CLI: https://github.com/Shopify/rubygems.org/pull/1
+
```

Comment:
Links are still a WIP - will be updated with better prototype links to look at within the next day or two